src/lib/loadInstructions.js

```diff
@@ -84,6 +164,56 @@ function slugifyExtension(ext) {
     .replace(/[^a-z0-9]+/g, "-")
     .replace(/^-+|-+$/g, "") || "unknown";
 }
+/*
+Example of expanding multi-segment fields into individual contiguous entries.
+
+Before expansion:
+fields = [
+  { label: "imm", from: 31, to: 7, segments: [ { from: 31, to: 25 }, { from: 11, to: 7 } ] },
+]
+
+After expansion:
+expanded = [
+  { label: "imm", from: 31, to: 25 },
+  { label: "imm", from: 11, to: 7 }
+]
+
+This ensures that each disjoint bit range (like imm[31:25] and imm[11:7])
+is rendered as its own segment in the bitfield diagram rather than one
+continuous bar spanning bits 31–7.
+*/
+
+function expandBitfieldFields(fields, totalBits = 32) {
+  const expanded = [];
+
+  // Expand multi-segment fields into individual entries
+  for (const field of fields) {
+    const segments = Array.isArray(field.segments) && field.segments.length
+      ? field.segments
+      : [{ from: field.from, to: field.to }];
+
+    for (const seg of segments) {
+      const hi = seg.from;
+      const lo = seg.to;
+      const width = hi - lo + 1;
+      if (width <= 0) continue;
+      const label =
+        segments.length > 1
+          ? `${field.label}`
+          : field.label;
+      expanded.push({
+        label,
+        from: hi,
+        to: lo,
```

Comment:
```suggestion
        from: seg.from,
        to: seg.to,
```

I think this would be more readable without the temporary fields.